packages/@react-aria/grid/src/useGridSelectionAnnouncement.ts

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright 2022 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import {announce} from '@react-aria/live-announcer';
+import {Collection, Node, Selection} from '@react-types/shared';
+// @ts-ignore
+import intlMessages from '../intl/*.json';
+import {Key, useRef} from 'react';
+import {SelectionManager} from '@react-stately/selection';
+import {useMessageFormatter} from '@react-aria/i18n';
+import {useUpdateEffect} from '@react-aria/utils';
+
+interface useGridSelectionAnnouncementProps {
+  /**
+   * A function that returns the text that should be announced by assistive technology when a row is added or removed from selection.
+   * @default (key) => state.collection.getItem(key)?.textValue
+   */
+  getRowText?: (key: Key) => string
+}
+
+interface GridSelectionState<T> {
+  /** A collection of items in the grid. */
+  collection: Collection<Node<T>>,
+  /** A set of items that are disabled. */
+  disabledKeys: Set<Key>,
+  /** A selection manager to read and update multiple selection state. */
+  selectionManager: SelectionManager
+}
```

Comment:
A more generalized state type than `GridState` since we actually provide a state of type `ListState` from useList. I could make this `GridState<T, GridCollection<T>>` but would then need to do some typescript hackiness (read: `as any`) in useList's call of useGridSelectionAnnouncement